src/Node/Expression/BlockReferenceExpression.php

```diff
@@ -25,7 +25,7 @@ class BlockReferenceExpression extends AbstractExpression
     /**
      * @param Node|null $template
      */
-    public function __construct(\Twig_NodeInterface $name, $template = null, $lineno, $tag = null)
+    public function __construct(\Twig_NodeInterface $name, $template = null, $lineno = 0, $tag = null)
```

Comment:
The right fix here would be to remove the `= null` on `$template` instead, keeping the same signature (where template and lineno must be provided).